### DISCUSSION TITLE
Add support for auto-delete of branches

### DISCUSSION THREAD

**Discussion by epage** [repository-settings/app#467] [API support] [answered]
## Prerequisites:

![Screenshot](https://docs.github.com/assets/images/help/repository/automatically-delete-branches.png)

* [Github Docs](https://docs.github.com/en/github/administering-a-repository/configuring-pull-request-merges/managing-the-automatic-deletion-of-branches)
* I couldn't find API availability - however I'm not that familiar with GitHub API docs so it might be.

## New Feature

Request: A config field allowing the repository settings "Automatically delete head branches " to be controlled.

**Comment by travi** [comment, marked as answer]
have you tried `delete_branch_on_merge` as mentioned in the example in the readme? that should be the attribute associated with this

**Reply by epage** [reply]
Looks like I missed it because I was using the example from https://github.com/apps/settings rather than https://github.com/probot/settings

**Reply by travi** [reply]
no problem. we'll work on getting that updated

when in doubt, but sure to also check the linked api docs, like https://docs.github.com/en/rest/reference/repos#update-a-repository for the repository section. that provides the full list of options, which may be more than we've specifically called out in our docs for the app
